decoder_borsh.go

```diff
@@ -86,6 +86,10 @@ func (dec *Decoder) decodeBorsh(rv reflect.Value, opt *option) (err error) {
 	}
 
 	rt := rv.Type()
+	if isTypeBorshEnumEmpty(rt) {
+		// Empty enum type, nothing to deserialize
+		return
+	}
```

Comment:
If you implement the `MarshalWithEncoder` method on `EmptyVariant` (see suggestion end of page), this check will not be necessary.

```suggestion
```